src/Artifacts.jl

```diff
@@ -226,6 +226,8 @@ function bind_artifact!(artifacts_toml::String, name::String, hash::SHA1;
         open(artifacts_toml, "w") do io
             TOML.print(io, artifact_dict, sorted=true)
         end
```

Comment:
Ah, these lines are buggy, causing this failure. Pkg must never open an existing file for writing, as it causes problems for the filesystem.

Reply:
e.g. https://github.com/JuliaLang/julia/issues/36664

Reply:
and https://github.com/JuliaLang/julia/issues/39892

Reply:
I think https://github.com/JuliaLang/julia/pull/46944 removed the check that was intended to caught things like this.

Reply:
It did, but that check was fundamentally flawed, and didn't actually check for things like this, it merely checked for attempts to *test for* this problem.